021-server-defined-attributes.md

```diff
@@ -0,0 +1,120 @@
+# Optional server-defined attributes
+
+* Creator: Khushboo Verma
+* Relevant Issues: N/A
+
+- [RFC Template](#rfc-template)
+  - [Summary](#summary)
+  - [Resources](#resources)
+  - [Implementation](#implementation)
+    - [API Changes](#api-changes)
+    - [Workers / Commands](#workers--commands)
+    - [Supporting Libraries](#supporting-libraries)
+    - [Data Structures](#data-structures)
+    - [SDKs](#sdks)
+    - [Breaking Changes](#breaking-changes)
+    - [Documentation & Content](#documentation--content)
+  - [Reliability](#reliability)
+    - [Security](#security)
+    - [Scaling](#scaling)
+    - [Benchmarks](#benchmarks)
+    - [Tests (UI, Unit, E2E)](#tests-ui-unit-e2e)
+  - [Open Questions](#open-questions)
+  - [Future Possibilities](#future-possibilities)
+
+## Summary
+<!-- Describe the problem we want to solve and suggested solution in a few paragraphs -->
+This RFC proposes the addition of optional server-defined attributes in all the project collections that can only be added by server and cannot be modified by the client. The user can add them to each collection through the API or console. Some useful sttributes can be `createdBy`, `updatedBy`, `ip` etc.
+
+## Resources
+<!-- List all the resources used for writing this RFC -->
+NA
+
+## Implementation
+
+<!-- Write an overview to explain the suggested implementation -->
+#### Console Changes
+We need to add an option in the `create attribute` settings dropdown to add a certain server-defined attribute from one of the existing options like `createdBy`.
```

Comment:
Hmm I was imagining a toggle. Having it as option in create attribute, it sounds like it's attribute type. Its like I can have multiple, of the same type. But in reality I can only have one `$createdBy`. Also if attribute creation modal, it would feel weird that there is name input but its disabled.

Reply:
Discussed as well. We won't allow enabling/ disabling attributes, so we don't need a toggle.